components/core/src/clp/NetworkReader.hpp

```diff
@@ -242,11 +247,19 @@ class NetworkReader : public ReaderInterface {
          * @param reader
          * @param offset Index of the byte at which to start the download.
          * @param disable_caching Whether to disable caching.
+         * @param http_header_kv_pairs Key-value pairs representing HTTP headers to pass to the
+         * server in the download request. Doc: https://curl.se/libcurl/c/CURLOPT_HTTPHEADER.html
          */
-        DownloaderThread(NetworkReader& reader, size_t offset, bool disable_caching)
+        DownloaderThread(
+                NetworkReader& reader,
+                size_t offset,
+                bool disable_caching,
+                std::optional<std::unordered_map<std::string, std::string>> http_header_kv_pairs
+        )
                 : m_reader{reader},
                   m_offset{offset},
-                  m_disable_caching{disable_caching} {}
+                  m_disable_caching{disable_caching},
+                  m_http_header_kv_pairs{http_header_kv_pairs} {}
```

Comment:
```suggestion
                  m_http_header_kv_pairs{std::move(http_header_kv_pairs)} {}
```
We should transfer the ownership to reduce memory copy